src/components/clusterDetail/ClusterInstallationProgressCard.tsx

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import { Card, Title, CardBody, CardHeader, CardExpandableContent } from '@patternfly/react-core';
+import { Cluster, Credentials, getClusterCredentials, stringToJSON } from '../../api';
+import { AlertsContextProvider } from '../AlertsContextProvider';
+import CancelInstallationModal from './CancelInstallationModal';
+import ClusterCredentials from './ClusterCredentials';
+import ClusterInstallationError from './ClusterInstallationError';
+import ClusterProgress from './ClusterProgress';
+import FailedHostsWarning from './FailedHostsWarning';
+import ResetClusterModal from './ResetClusterModal';
+import HostsTable from '../hosts/HostsTable';
+import ClusterDetailsButtonGroup from './ClusterDetailsButtonGroup';
+import { ClusterStatusIcon } from '../clusters/ClusterStatus';
+
+import './ClusterInstallationProgressCard.css';
+
+type AIClusterDetailsCardProps = {
+  clusterObj: Cluster;
+};
+
+const getID = (suffix: string) => `cluster-detail-${suffix}`;
+
+const AIClusterDetailsCard: React.FC<AIClusterDetailsCardProps> = ({ clusterObj }) => {
+  // CamelCase the cluster object
+  const cluster: Cluster = stringToJSON<Cluster>(JSON.stringify(clusterObj)) || clusterObj;
```

Comment:
IIUC, the passed `clusterObj` is not of the `Cluster` type as defined here.
Same comment as in case of the `ClusterNetworDetails` component bellow.